### PR TITLE
Adding USB Support for dive computers

### PR DESCRIPTION
**Pull request by Ben1980**
Provides USB and USB HID support for macOS.
All dependencies come via git submodules and are built during example compilation time.
Logging is also now working but needs a review.
I was only able to confirm USB HID working with Suunto EON Core, can't say if USB is working but I guess it is.

### CODE REVIEW

**Comment by Ben1980** [comment]
@Sese-Schneider I'm not sure if the force-pushed was a good idea. I have now 275 different files. I don't think you need to update my working branch.

File: example/macos/Runner/Info.plist [hunk 1]
```diff
@@ -28,5 +28,7 @@
 	<string>MainMenu</string>
 	<key>NSPrincipalClass</key>
 	<string>NSApplication</string>
+	<key>NSUSBPeripheralUsageDescription</key>
+    <string>Your explanation here</string>
```
Comment (Sese-Schneider) [inline example/macos/Runner/Info.plist]: Formatting
```suggestion
        <string>Your explanation here</string>
```

File: lib/framework/interfaces/dive_computer_interfaces.dart [hunk 2]
```diff
@@ -6,26 +6,26 @@ import '../utils/utils.dart';
...
       case ComputerTransport.serial:
         return _connectSerial(computer);
-      // case ComputerTransport.usbhid:
-      //   return _connectUsbHid(computer);
-      case ComputerTransport.ble:
-        return _connectBle(computer);
+      case ComputerTransport.usb:
+        return _connectUsb(computer);
+      case ComputerTransport.usbhid:
+        return _connectUsbHid(computer);
```
Comment (Sese-Schneider) [inline lib/framework/interfaces/dive_computer_interfaces.dart:28]: Do you now know the difference between the USB and USBHID in `libdivecomputer`?

File: macos/Classes/hid.c
Comment (Sese-Schneider) [inline macos/Classes/hid.c:1]: is this file still required, now that you've removed all others?

**Review by Sese-Schneider — CHANGES_REQUESTED** [review]
Some small things you could optimize.
In general I love this! Great work and thank you so much 🚀

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -111,11 +168,12 @@ add_library(dive_computer SHARED
...
 
-include_directories("libdivecomputer/include")
+target_link_libraries(libdivecomputer PRIVATE hidapi::hidapi)
+target_link_libraries(libdivecomputer PRIVATE libusb)
 
-set_target_properties(dive_computer PROPERTIES
+set_target_properties(libdivecomputer PROPERTIES
   PUBLIC_HEADER dive_computer.h
-  OUTPUT_NAME "dive_computer"
+  OUTPUT_NAME "libdivecomputer"
```
Comment (Sese-Schneider) [inline src/CMakeLists.txt:176]: Why does the output name need to be `libdivecomputer`?

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -3,9 +3,63 @@
...
+add_compile_definitions(PACKAGE_STRING="libdivecomputer 0.9.0-devel")
+add_compile_definitions(PACKAGE_TARNAME="libdivecomputer")
+add_compile_definitions(PACKAGE_URL="")
+add_compile_definitions(PACKAGE_VERSION="0.9.0-devel")
+add_compile_definitions(STDC_HEADERS=1)
+add_compile_definitions(VERSION="0.9.0-devel")
+
+# add_compile_definitions(__APPLE__=1)
+# add_compile_definitions(HAVE_HIDAPI=1)
+# add_compile_definitions(HAVE_PTHREAD_H=1)
```
Comment (Sese-Schneider) [inline src/CMakeLists.txt:54]: Can this be removed?

File: pubspec.yaml [hunk 5]
```diff
@@ -12,18 +12,18 @@ environment:
...
-  logging: '>=1.2.0 <2.0.0'
-  path: '>=1.8.0 <2.0.0'
-  plugin_platform_interface: '>=2.0.2 <3.0.0'
+  logging: ^1.2.0
+  path: ^1.8.3
+  plugin_platform_interface: ^2.1.8
 
 dev_dependencies:
-  ffigen: ">=10.0.0 <12.0.0"
+  ffigen: ^9.0.0
```
Comment (Sese-Schneider) [inline pubspec.yaml]: Why are you downgrading ffigen?

File: pubspec.yaml
Comment (Sese-Schneider) [inline pubspec.yaml:1]: Is there a reason you're fixing the constraints to specific versions here?

**Review by Sese-Schneider — COMMENTED** [review]
Had to leave for work, here's my finalized review

**Comment by Ben1980** [comment]
@Sese-Schneider I think we should discuss if this PR can be merged or how changes should look like. Shall we have a call?

**Comment by Sese-Schneider** [comment]
@Ben1980 PR looks very good so far - I want to test it myself after the weekend and merge it ASAP.
I also want to give Windows another try later.
Nevertheless, I try to get it to main and publish this version hopefully soon.
If I have more questions we can call, or if you want to anyways to talk about it send me an invite please 👍🏼

**Comment by Sese-Schneider** [comment]
merging to develop - windows is still not working so this would be my next step to get it to main.
I will publish this as `v0.1.0-dev.1`

**Pull request merged**